### PR TITLE
Compatibility for compositional data

### PR DESCRIPTION
**Pull request by marlonsmathias**

### CODE REVIEW

**Comment by juliohm** [comment]
That is awesome, we just need a simple test to make sure that future changes won't break the feature. Can you change to the test directory and run `] add CoDa` to the test dependencies of the test folder and add a simple test at the end of the file with a table containing coda?

File: src/estimators/sk.jl [hunk 1]
```diff
@@ -43,7 +43,7 @@ function combine(fitted::FittedKriging{<:SimpleKriging},
   μ = fitted.estimator.μ
   b = fitted.state.RHS
   λ = weights.λ
-  y = z .- μ
+  y = map(z->z-μ,z)
```
Comment (juliohm) [inline src/estimators/sk.jl]: What is the error message with the original z .- mu code? I thought it should work fine with coda.

File: test/estimators.jl [hunk 2]
```diff
@@ -201,4 +201,36 @@
...
+
+  # ------------------
+  # COMPOSITIONAL DATA
+  # ------------------
+
+  # create some data
+  dim = 2; nobs = 10; elements = 4;
+  pset = PointSet(10*rand(dim, nobs));
+  #data = georef((z=[rand(elements) for r in 1:nobs],), pset)
+  data = georef((z=[Composition(rand(elements)) for r in 1:nobs],), pset);
```
Comment (juliohm) [inline test/estimators.jl]:
```suggestion
  dim = 2; nobs = 10
  pset = PointSet(10*rand(dim, nobs))
  table = (z=rand(Composition{3}, nobs),)
  data = georef(table, pset);
```

File: test/estimators.jl [hunk 3]
```diff
@@ -201,4 +201,36 @@
...
+  ordkrig = OrdinaryKriging(data, γ)
+  unikrig = UniversalKriging(data, γ, 1)
+  drikrig = ExternalDriftKriging(data, γ, [x->1.])
+
+  # prediction on a quadrangle
+  uₒ = Quadrangle((0.,0.), (1.,0.), (1.,1.), (0.,1.))
+  _, SKvar = predict(simkrig, :z, uₒ)
+  _, OKvar = predict(ordkrig, :z, uₒ)
+  _, UKvar = predict(unikrig, :z, uₒ)
+  _, DKvar = predict(drikrig, :z, uₒ)
```
Comment (juliohm) [inline test/estimators.jl]: We need to check that the returned estimate is a coda object, for example `@test coda isa Composition`

File: test/runtests.jl [hunk 4]
```diff
@@ -7,6 +7,7 @@ using Statistics
 using Plots; gr(size=(600,400))
 using ReferenceTests, ImageIO
 using Test, Random
+using CoDa
```
Comment (juliohm) [inline test/runtests.jl]: Please move this `using CoDa` to before `using LinearAlgebra`.

**Review by juliohm — CHANGES_REQUESTED** [review]
We need to add CoDa to the test dependencies. As I explained, you need to cd to the test folder, activate the folder with `] activate .` and then `] add CoDa` to the test dependencies. This will modify the Project.toml, which will be part of the PR.

File: src/estimators/sk.jl (hunk 1 above)
Comment (marlonsmathias) [reply src/estimators/sk.jl]: The error happened due to trying to subtract an array com compositions by a composition:
`Estimators: Error During Test at c:\Users\Marlon\.julia\dev\KrigingEstimators\test\estimators.jl:1
  Got exception outside of a @test
  MethodError: no method matching length(::Composition{3, (:w1, :w2, :w3)})
  Closest candidates are:
    length(::Union{Base.KeySet, Base.ValueIterator}) at abstractdict.jl:58
    length(::Union{Adjoint{T, var"#s6"} where var"#s6"<:Union{StaticArrays.StaticVector{var"#s1", T} where var"#s1", StaticArrays.StaticMatrix{var"#s4", var"#s5", T} where {var"#s4", var"#s5"}}, Diagonal{T, var"#s13"} where var"#s13"<:(StaticArrays.StaticVector{var"#s14", T} where var"#s14"), Hermitian{T, var"#s10"} where var"#s10"<:(StaticArrays.StaticMatrix{var"#s11", var"#s12", T} where {var"#s11", var"#s12"}), LowerTriangular{T, var"#s18"} where var"#s18"<:(StaticArrays.StaticMatrix{var"#s19", var"#s20", T} where {var"#s19", var"#s20"}), Symmetric{T, var"#s7"} where var"#s7"<:(StaticArrays.StaticMatrix{var"#s8", var"#s9", T} where {var"#s8", var"#s9"}), Transpose{T, var"#s1"} where var"#s1"<:Union{StaticArrays.StaticVector{var"#s1", T} where var"#s1", StaticArrays.StaticMatrix{var"#s4", var"#s5", T} where {var"#s4", var"#s5"}}, UnitLowerTriangular{T, var"#s24"} where var"#s24"<:(StaticArrays.StaticMatrix{var"#s25", var"#s26", T} where {var"#s25", var"#s26"}), UnitUpperTriangular{T, var"#s21"} where var"#s21"<:(StaticArrays.StaticMatrix{var"#s22", var"#s23", T} where {var"#s22", var"#s23"}), UpperTriangular{T, var"#s15"} where var"#s15"<:(StaticArrays.StaticMatrix{var"#s16", var"#s17", T} where {var"#s16", var"#s17"}), StaticArrays.StaticArray{var"#s26", T, N} where {var"#s26"<:Tuple, N}, StaticArrays.StaticVector{var"#s26", T} where var"#s26", StaticArrays.StaticMatrix{var"#s5", var"#s4", T} where {var"#s5", var"#s4"}} where T) at C:\Users\Marlon\.julia\packages\StaticArrays\OWJK7\src\abstractarray.jl:1
    length(::Union{Adjoint{T, S}, Transpose{T, S}} where {T, S}) at C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\LinearAlgebra\src\adjtrans.jl:195`

Comment (juliohm) [reply src/estimators/sk.jl]: Got it, I will suggest a cleaner version then.

Comment (juliohm) [inline src/estimators/sk.jl]:
```suggestion
  y = [zᵢ - μ for zᵢ in z]
```

File: test/estimators.jl [hunk 5]
```diff
@@ -201,4 +201,42 @@
...
+
+  # ------------------
+  # COMPOSITIONAL DATA
+  # ------------------
+
+  # create some data
+  dim = 2; nobs = 10
+  pset = PointSet(10*rand(dim, nobs))
+  table = (z=rand(Composition{3}, nobs),)
+  data = georef(table, pset);
```
Comment (juliohm) [inline test/estimators.jl]:
```suggestion
  data = georef(table, pset)
```

File: test/estimators.jl [hunk 6]
```diff
@@ -201,4 +201,42 @@
...
+  @test OKvar + tol ≥ 0
+  @test UKvar + tol ≥ 0
+  @test DKvar + tol ≥ 0
+  @test SKvar ≤ OKvar + tol
+
+  # type tests
+  @test SKpred isa Composition
+  @test OKpred isa Composition
+  @test UKpred isa Composition
+  @test DKpred isa Composition
```
Comment (juliohm) [inline test/estimators.jl]:
```suggestion
  # type tests
  @test SKpred isa Composition
  @test OKpred isa Composition
  @test UKpred isa Composition
  @test DKpred isa Composition
  
  # variance checks
  @test SKvar + tol ≥ 0
  @test OKvar + tol ≥ 0
  @test UKvar + tol ≥ 0
  @test DKvar + tol ≥ 0
  @test SKvar ≤ OKvar + tol
```

File: test/estimators.jl [hunk 7]
```diff
@@ -201,4 +201,42 @@
...
+  
+  # variance checks
+  @test SKvar + tol ≥ 0
+  @test OKvar + tol ≥ 0
+  @test UKvar + tol ≥ 0
+  @test DKvar + tol ≥ 0
+  @test SKvar ≤ OKvar + tol
+
```
Comment (juliohm) [inline test/estimators.jl]:
```suggestion
```

**Review by juliohm — APPROVED** [review]

**Comment by juliohm** [comment]
Awesome. The build failure on Mac is because of our tight tolerances in some of the tests. I will go ahead and merge this.

**Pull request merged**